### ISSUE TITLE
Setting textSize in draw() function of Processing.js results in text disappearing (in Chrome only)

### ISSUE THREAD

**Issue by danielgregorywilson** [processing-js/processing-js#167]
The following processing.js code initially draws a background and text. However, a few seconds later, the text flashes quickly; and a few seconds after that, the text disappears entirely.

<pre><code>processing.draw = function() {
    processing.background(0,255,0);
    processing.textSize(20);
    processing.text("Test", 30, 30); 
};</code></pre>

The issue seems to be with the textSize() function. If you remove that line, the problem disappears. Here is my fiddle:

https://jsfiddle.net/bbh9r8bp/12/

Note that this seems isolated to Chrome, since it occurs with v41.0.2272.104 (64-bit), but neither on Safari v8.0.4 nor on Firefox v29.0.1. I am using Processing v1.4.8 in my project, but the issue is the same in the fiddle using v1.4.7. Earlier versions of the library seem to not have this issue (tried with v1.0.0 on jsfiddle and on <a href="http://khan.github.io/live-editor/demos/simple/">Khan Academy's live editor</a> which seems to have originated with Processing.js 1.2.3 or thereabouts.

This may well be an issue with the Processing library, but it seems like such a simple bug that I must be missing something.

**Comment by Pomax** [comment]
As per Processing rules, you can't really call `size` [except once](https://processing.org/reference/size_.html) (viz: "*The size() function can only be used once inside a sketch, and it cannot be used for resizing*"), in setup(). any other time it's going to do crazy bootstrapping that should never happen mid-sketch-run (both in Processing and Processing.js)

**Comment by teo1978** [comment]
@Pomax  I think he's talking about ```textSize()```, not ```size()```

**Comment by Pomax** [comment]
I am blind.

**Comment by Pomax** [comment]
I tried the following code, which seems to work correctly in Chrome (see http://jsbin.com/qaneqanuxe/1/edit?html,output)
```
      float t = 0;

      void setup() {
        size(300,300);
        textAlign(CENTER, CENTER);
      }

      void draw() {
        background(0,255,0);
        if(t>TAU) { t = 0; }
        t += 0.05;
        int b = 20;
        int r = int(b * sin(t));
        int s = 2*b + r;
        textSize(s);
        text("Test", width/2, height/2); 
      }
```
Note that the "pure JS" way of writing Processing.js code has been deprecated - using straight up JS will in most cases still work, but the power of Processing.js was not in its JS API (for which paper.js might be a better candidate) but in its automagically running of plain Processing code, so I'd recommend using this style of code instead (even for interfacing with non-sketch JS functions on the page. If they're globally available, the sketch will be able to use them).

**Comment by danielgregorywilson** [comment]
Yeah, your code definitely works. However, substituting mine still seems to produce the same issue:
http://jsbin.com/gozexixuxi/1/edit?html,output

    void setup() {
        size(300,300);
        textAlign(CENTER, CENTER);
      }

      void draw() {
        background(0,255,0);
        textSize(20);
        text("Test", 30, 30); 
      }

Am I just using textSize or background wrong? It just seems too simple to be incorrect.

**Comment by Pomax** [comment]
usually you only once to set textSize "once", so if I move it into the setup() function, for instance, things work fine. From a programming perspective: that's probably where it should be, but from a processing.js source code perspective: textSize(X) with X being what it's set to already should probably not do anything, so the fact that the text disappears in Chrome is really, really weird. Worth looking into!

**Comment by danielgregorywilson** [comment]
A recent chrome update seems to have resolved the issue (testing on Chrome v42.0.2311.90).

**Comment by Pomax** [comment]
hmn, chalking this up as a wonky Chrome thing, then. But if it happens still, we can reopen (or refile)
